### DISCUSSION TITLE
Is it possible to use two data sources or indexes in Semantic Kernel?

### DISCUSSION THREAD

**Discussion by Cobra86** [microsoft/semantic-kernel#6703] [Q&A] [answered]
I am trying to use two indexes in a single Azure AI Search instance. I attempted to add two extensions, but I keep receiving an error indicating that only one data source can be defined.

Is it possible to use two data sources or indexes in Semantic Kernel?

**Comment by sophialagerkranspandey** [comment]
The limitation of the current connector is you can only have one data source with AzureAISearch

**Comment by jamesmkfoo23** [comment]
@Cobra86 , when you say two **extensions**, do you mean the Azure Open AI Service /extensions API with **dataSource** parameter?

I am wondering is it better to just use the generic /completions API and inject the Azure AI Search retrieval content manually?

**Comment by dmytrostruk** [comment, marked as answer]
Since `index` is configured on per-request basis, I think it should be possible to use multiple indexes for different requests:
https://github.com/microsoft/semantic-kernel/blob/8033515a86964a6b712886b0910d5b5fa8accfe7/dotnet/samples/Concepts/ChatCompletion/AzureOpenAIWithData_ChatCompletion.cs#L50-L55

But I'm not sure it's possible to specify multiple indexes as part of one request.

**Comment by Cobra86** [comment]
I was able to switch between two different Azure AI services at runtime, but I was doing it manually, allowing the user to toggle between the services.

I'm not sure if it's good practice to create Kernel plugins and then call chatCompletion.GetChatMessageContentAsync within the plugin function.

**Comment by Cobra86** [comment]
Also, I can't use Azure AI search extension and kernel plugins together.

**Reply by dmytrostruk** [reply]
This is a known limitation:
https://github.com/Azure/azure-sdk-for-net/tree/49b0ace48d35cb4b2ace93b517036b660fc12ca4/sdk/openai/Azure.AI.OpenAI#use-your-own-data-with-azure-openai
> NOTE: The concurrent use of [Chat Functions](https://github.com/Azure/azure-sdk-for-net/tree/49b0ace48d35cb4b2ace93b517036b660fc12ca4/sdk/openai/Azure.AI.OpenAI#use-chat-functions) and Azure Chat Extensions on a single request is not yet supported. Supplying both will result in the Chat Functions information being ignored and the operation behaving as if only the Azure Chat Extensions were provided. To address this limitation, consider separating the evaluation of Chat Functions and Azure Chat Extensions across multiple requests in your solution design.

**Reply by Cobra86** [reply]
Thank you.
